.github/scripts/transform_dbt_test_results.py

```diff
@@ -676,28 +621,119 @@ def _filter_for_existing_field_indexes(
+  3. `metadata/test_run_result/run_year=YYYY/*.parquet`: Metadata about test results (pass, fail, number of failing rows, etc.) in this run,
+     partitioned by year of run and prepped for upload to S3
+
+Each sheet in the output workbook represents a category of test, e.g. "valid_range" or "not_null"; each row in a sheet represents a row in
+a database that failed a test, with enough metadata that a reader can figure out what conditions caused the test to fail and investigate the root cause."""  # noqa: E501
+
+# Examples to use in the command line interface docstring
+CLI_EXAMPLE = """Example usage with no options provided:
+
+    python3 transform_dbt_test_results.py
+
+Example usage with all options provided:
+
+    AWS_ATHENA_S3_STAGING_DIR=s3://foo-bar-baz/ python3 transform_dbt_test_results.py
+        --run-results ./target/run_results.json
+        --manifest ./target/manifest.json
+        --output-dir ./qc_test_results/
+        --township 77
+
+Example usage to filter for multiple townships:
+
+    python3 transform_dbt_test_results.py --township 76 77"""  # noqa: E501
+
+
 def main() -> None:
     """Entrypoint to this script. Parses dbt test results and writes artifacts
     to the output directory with metadata about tests."""
-    try:
-        run_results_filepath = sys.argv[1]
-    except IndexError:
-        run_results_filepath = os.path.join("target", "run_results.json")
 
-    try:
-        manifest_filepath = sys.argv[2]
-    except IndexError:
-        manifest_filepath = os.path.join("target", "manifest.json")
+    parser = argparse.ArgumentParser(
+        description=CLI_DESCRIPTION,
+        epilog=CLI_EXAMPLE,
+        # Parse the description and epilog as raw text so that newlines
+        # get preserved
+        formatter_class=argparse.RawTextHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--run-results",
+        required=False,
+        default=os.path.join("target", "run_results.json"),
+        help=(
+            "The local path to a run_results.json file generated by a "
+            "`dbt test --store-failures` run. Defaults to "
+            "'./target/run_results.json'."
+        ),
+    )
+    parser.add_argument(
+        "--manifest",
+        required=False,
+        default=os.path.join("target", "manifest.json"),
+        help=(
+            "The local path to a manifest.json file with the compiled dbt "
+            "DAG, generated by all dbt CLI commands (including `dbt test`). "
+            "Defaults to './target/manifest.json'."
+        ),
+    )
+    parser.add_argument(
+        "--output-dir",
+        required=False,
+        help=(
+            "The directory to which output artifacts should be written; "
+            "if the directory does not exist, it will be created. Defaults to "
+            "'./qc_test_results_<date>/'."
+        ),
+    )
+    parser.add_argument(
+        "--township",
+        required=False,
+        nargs="*",
```

Comment:
Setting this option instructs argparse to accept one or more values for this option and combine them into a tuple.